File: DI/DI-Dagger-Java/src/main/java/model/presenter/MyView.java
package model.presenter;

import DI.presenter.DaggerGrafoDependencias;

import javax.inject.Inject;
import java.util.UUID;

public class MyView {
    private final String id = UUID.randomUUID().toString();
    @Inject
    Presenter presenter;

    // Aquí no hacemos la inyección en el constructor si no más adelante
    // por eso necesitamos el metodo inject
    // Esto es interesante en sitios donde no hay cosntructores
    // Como Android y sus actividades
    public MyView() {
        init();
    }

    public void init() {
        DaggerGrafoDependencias.create().inject(this);
    }

    @Override
    public String toString() {
        return "MyView{" +
                "presenter=" + presenter +
                ", id='" + id + '\'' +
                '}';
    }
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/models/presenter/MyView.kt
package models.presenter

import DI.presenter.DaggerGrafoDependencias
import java.util.*
import javax.inject.Inject


class MyView {
    private val id = UUID.randomUUID().toString()

    @Inject
    lateinit var presenter: Presenter

    // Aquí no hacemos la inyección en el constructor si no más adelante
    // por eso necesitamos el metodo inject
    // Esto es interesante en sitios donde no hay cosntructores
    // Como Android y sus actividades
    init {
        DaggerGrafoDependencias.create().inject(this)
    }
    
    override fun toString(): String {
        return "MyView{" +
                "presenter=" + presenter +
                ", id='" + id + '\'' +
                '}'
    }
}
